packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -410,7 +429,9 @@ class _HeroFlight {
       assert(type != null);
       switch (type) {
         case HeroFlightDirection.pop:
-          return initial.value == 1.0 && initial.status == AnimationStatus.reverse;
+          // Pops may be initiated by a user gesture that manually drives the
```

Comment:
Was there a way to for initial.value == 1.0 and initial.status != AnimationStatus.reverse absent a user triggered transition? It looks like we've dropped that now.

Reply:
I'm not sure I understood. Can you rephrase what you meant?

Reply:
I have hoping to keep the gesture concept out of the flight and the manifests. Added the assert back.